functions/go/fix/fixpkg/fix.go

```diff
@@ -230,15 +230,18 @@ func (s *Fix) FunctionsInPkg(nodes []*yaml.RNode, pkgPath string) []v1.Function
 		}
 		fnSpec := runtimeutil.GetFunctionSpec(node)
 		if fnSpec != nil {
+			// in order to make sure there is only one function config per file,
+			// rename the file with the name(and hash(name)) of the function config resource
+			fnFileName := fmt.Sprintf("%s.yaml", node.GetName())
```

Comment:
nodeName can be empty also ? (We have added KRM checks recently, and other day we found `live apply` also running into this edge case).

Reply:
kpt fn eval errors out if there is no name. So the function will not be invoked. If we get to this point, it means the name already exists.